CMakeLists.txt

```diff
@@ -292,10 +292,10 @@ endif()
 
 if(WITH_OTLP)
   set(protobuf_MODULE_COMPATIBLE ON)
-  find_package(Protobuf)
+  find_package(Protobuf PATHS ${EXTERNAL_PROTOBUF_DIR})
```

Comment:
Why we need this? External protobuf and grpc can be found by add installation path into [CMAKE_FIND_ROOT_PATH](https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_ROOT_PATH.html#variable:CMAKE_FIND_ROOT_PATH) and [CMAKE_MODULE_PATH ](https://cmake.org/cmake/help/latest/variable/CMAKE_MODULE_PATH.html#variable:CMAKE_MODULE_PATH),or by setting `Protobuf_DIR/Protobuf_ROOT` and `gRPC_DIR/gRPC_ROOT` ?

> See https://cmake.org/cmake/help/latest/command/find_package.html#id4 for details

Reply:
Thanks @owent, I used your [suggestion](https://github.com/open-telemetry/opentelemetry-cpp/issues/1384#issuecomment-1120574543) on the issue, and it works without my changes.
I have another issue now. The files for this library https://github.com/open-telemetry/opentelemetry-cpp/blob/54abc2741b10efef07f786bb5200fd9112b191bb/cmake/opentelemetry-proto.cmake#L191-L203 are not available when calling make, so the first make try fails and if I repeat it the build goes through. Do you have any suggestion to solve this issue?
otherwise I have to close this PR.

Reply:
Sorry, I don't really get the point about the files are not available when calling make. These files are generated by `add_custom_command(OUTPUT [files]` above.
https://github.com/open-telemetry/opentelemetry-cpp/blob/54abc2741b10efef07f786bb5200fd9112b191bb/cmake/opentelemetry-proto.cmake#L163
This command should run when OUTPUT files are expired or not found and when they are depended to build any target.
If you are writing a standalone Makefile, I think it's required to write a similar TARGET of Makefile to generate these files.

Reply:
The problem is that, the files are not there and custome_command is ran on build time not config time. on build time when I use 16 cores to build most of the times the PB files are not generated before the compile of proto library. I think this can be solved by using execute_process.

Reply:
Why we need the generated files at config time?If we want a standalone target to just generate these files, but not compile them. I think the better way is adding a custom target by `add_custom_target(generate_opentelemetry_proto SOURCES [source files...])` and run `cmake --build . --target generate_opentelemetry_proto` after configure is done.

Reply:
thanks @owent. I'm going to close this PR.